eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -1,61 +1,32 @@
-parameters:
-  # Use the variable name itself (no $() or other wrapping syntax). This is
-  # because we use runtime and macro expressions for conditions and script
-  # invocations
-  DependenciesVariableName: vcpkg.deps
-
 steps:
   - pwsh: |
       $TargetPath = "$(Agent.TempDirectory)/vcpkg"
       Remove-Item -Path $TargetPath -Recurse -Force -ErrorAction Ignore
       New-Item -ItemType Directory -Path $TargetPath  -Force
 
       $VcpkgCommit = $(Get-Content eng/vcpkg-commit.txt)
+      $VcpkgCacheMode = If (Test-Path 'env:VCPKG_CACHE_SAS') {"readwrite"} Else {"read"}
       Write-Host "Target Path for vcpkg: $TargetPath"
       Write-Host "Vcpkg SHA: $VcpkgCommit"
+      Write-Host "Cache mode: $env:VCPKG_CACHE_MODE"
 
       Write-Host "##vso[task.prependpath]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VcpkgCommit]$VcpkgCommit"
+      Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES;issecret=true]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,$env:VCPKG_CACHE_SAS,$VcpkgCommit"
     displayName: Set Vcpkg Variables
+    env:
+      VCPKG_CACHE_SAS: $(cpp-vcpkg-cache-write-sas)
```

Comment:
You should put an if around this to only try and access this sas token for internal builds.